### ISSUE TITLE
Cancelled purchase is still added to the Total in Purchase history

### ISSUE THREAD

**Issue by superm1991** [ChristopherKlay/StadiaEnhanced#111]
Hi,

I've actually bought games and Pro sub for a total of around 160€, but the extension shows around 220€ because Cyberpunk 2077 is counted twice (my first purchase was cancelled because of a credit card issue).

Here is the first (cancelled) occurrence:
```html
<div class="l6cQMc" jsaction="..." data-transaction-id="...">
	<div class="h6J22d undefined QAAyWd" tabindex="0">
		<div class="lYyTnf">
			<div style="background-image:url(https://lh3.googleusercontent.com/LYrpD92A8ND315aew206bpXFLw3nsMZsgOpoMgRUGYFCfRrsDmKGejGG99LK3uVMssLNT_PlFeuORdkC3ixRkT03p9WS-iVWURbEpIsZ63V2Q2j_CkIdmQtJKQ=w214-h120-e14-rw-no)" class="WDI43c"></div>
		</div>
		<div class="zRamU Ky7Jwc">
			<div class="Mo7he iOzlzc">
				<div class="T2oslb zHGix">Cyberpunk 2077</div>
				<div class="ilWqpe jwCbh">
					<div class="lT8jMd">
						<div class="v3rApc KAiU7c">Nov 19, 2020</div>
						<div class="v3rApc Wh3gGb">Cancelled</div>
					</div>
				</div>
			</div>
		</div>
		<div class="qLWGEf Pf3n9d">
			<div class="o2NwGe">€59.99</div>
		</div>
		<div class="QL3BFc"></div>
	</div>
</div>
```

Here is the second (actual purchase) occurrence:
```html
<div class="l6cQMc" jsaction="..." data-transaction-id="...">
	<div class="h6J22d undefined QAAyWd" tabindex="0">
		<div class="lYyTnf">
			<div style="background-image:url(https://lh3.googleusercontent.com/LYrpD92A8ND315aew206bpXFLw3nsMZsgOpoMgRUGYFCfRrsDmKGejGG99LK3uVMssLNT_PlFeuORdkC3ixRkT03p9WS-iVWURbEpIsZ63V2Q2j_CkIdmQtJKQ=w214-h120-e14-rw-no)" class="WDI43c"></div>
		</div>
		<div class="zRamU Ky7Jwc">
			<div class="Mo7he iOzlzc">
				<div class="T2oslb zHGix">Cyberpunk 2077</div>
				<div class="ilWqpe jwCbh">
					<div class="lT8jMd">
						<div class="v3rApc KAiU7c">Dec 10, 2020</div>
					</div>
				</div>
			</div>
		</div>
		<div class="qLWGEf Pf3n9d">
			<div class="o2NwGe">€59.99</div>
		</div>
		<div class="QL3BFc"></div>
	</div>
</div>
```

**Comment by ChristopherKlay** [comment]
Just to make sure, since i sadly don't have any refunds on my account:

`document.querySelectorAll(".v3rApc.Wh3gGb")` only highlight the refunded ones, correct?

**Comment by ChristopherKlay** [comment]
I'll get it checked on some other accounts until tomorrow and then implement a fix that filters those entries out for the next patch that should be out in a day or two (:

**Comment by ChristopherKlay** [comment]
Turns out; This won't actually work.

You can identify refunded items with this, however; the filter also hits pre-orders, that are tagged via the same class name.

I will have to look into finding either a different way to identify refunded items, or a different way to read/collect the data in the first place, since currently the only difference between the two is a text string that changes with every single possible supported language.

**Comment by superm1991** [comment]
Maybe something like this?
- An item that appears in the list exactly once and without ".v3rApc.Wh3gGb" is a regular purchase.
- An item that appears in the list exactly once and with ".v3rApc.Wh3gGb" is a pre-order.
- An item that appears in the list exactly twice, first without ".v3rApc.Wh3gGb" then with ".v3rApc.Wh3gGb" is a regular purchase that was cancelled.
- An item that appears in the list exactly twice, first with ".v3rApc.Wh3gGb" then again with ".v3rApc.Wh3gGb" is a pre-order that was cancelled.
- etc...

Basically: any item that appears with ".v3rApc.Wh3gGb" will cancel the previous occurrence of the same item (therefore it won't cancel anything if it's a pre-order because that would be the first occurrence).

Note: I think refunds are only available once per item, and we can't buy an item that we already own, so we shouldn't ever see the same item appear more than 3 times.

**Comment by ChristopherKlay** [comment]
Fixed this a while ago - forgot to close this.

My bad!
